src/main/java/uk/gov/digital/ho/hocs/domain/queue/migration/MigrationService.java

```diff
@@ -38,40 +44,121 @@ public class MigrationService {
+        try {
+            var migrationCorrespondentRequest  = composeMigrationCorrespondentRequest(
+                    caseId,
+                    stageId,
+                    migrationCaseData);
+            ResponseEntity responseEntity = migrationCaseworkClient.migrateCorrespondent(migrationCorrespondentRequest);
+            messageLogService.updateCaseUuidAndStatus(requestData.getCorrelationId(), caseId, Status.CASE_CORRESPONDENTS_HANDLED);
+            log.info("Created correspondents for migrated case {}", caseId);
+            return responseEntity;
+        } catch (Exception e) {
+            messageLogService.updateStatus(requestData.getCorrelationId(), Status.CASE_CORRESPONDENTS_FAILED);
+            throw new ApplicationExceptions.CaseCorrespondentCreationException(e.getMessage(), LogEvent.CASE_CORRESPONDENTS_FAILURE);
+        }
+    }
+
+    ResponseEntity createCaseAttachments(UUID caseId, MigrationData migrationCaseData) {
+        try {
+            var caseAttachments = getCaseAttachments(
+                    caseId,
+                    migrationCaseData);
+            for(CaseAttachment attachment : caseAttachments) {
+                CreateDocumentRequest document =
+                        new CreateDocumentRequest(
+                                attachment.getDisplayName(),
+                                attachment.getType(),
+                                attachment.getDocumentPath(),
+                                caseId);
+                documentClient.createDocument(caseId, document);
+            }
+            messageLogService.updateCaseUuidAndStatus(requestData.getCorrelationId(), caseId, Status.CASE_DOCUMENT_CREATED);
+            log.info("Created case attachments for migrated case {}", caseId);
+        } catch (Exception e) {
+            messageLogService.updateStatus(requestData.getCorrelationId(), Status.CASE_DOCUMENT_FAILED);
+            throw new ApplicationExceptions.DocumentCreationException(e.getMessage(), LogEvent.CASE_DOCUMENT_CREATION_FAILURE);
+        }
+        return ResponseEntity.ok().build();
     }
 
     CreateMigrationCaseRequest composeMigrateCaseRequest(MigrationData migrationData, MigrationCaseTypeData migrationCaseTypeData) {
         Map<String, String> initialData = Map.of(CHANNEL_LABEL, migrationCaseTypeData.getOrigin());
 
-        MigrationComplaintCorrespondent primaryCorrespondent = getPrimaryCorrespondent(migrationData.getPrimaryCorrespondent());
-        List<MigrationComplaintCorrespondent> additionalCorrespondents = getAdditionalCorrespondents(migrationData.getAdditionalCorrespondents());
-        List<CaseAttachment> caseAttachments = getCaseAttachments(migrationData.getCaseAttachments());
-
         return new CreateMigrationCaseRequest(
                 migrationData.getComplaintType(),
                 migrationData.getDateReceived(),
                 initialData,
-                "MIGRATION",
+                "MIGRATION");
```

Comment:
Is this the right stage type? in hocs-data there are case type specific stage names I think?